util/log: TestRedactedDecodeFile failed

util/log.TestRedactedDecodeFile [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3978479&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3978479&tab=artifacts#/) on master @ [8661fdc034a6e787c07d7a9655b9241745d2ef33](https://github.com/cockroachdb/cockroach/commits/8661fdc034a6e787c07d7a9655b9241745d2ef33):


```
      /usr/local/go/src/bufio/bufio.go:604 +0x5e
  github.com/cockroachdb/cockroach/pkg/util/log.(*syncBuffer).Flush()
      <autogenerated>:1 +0x49
  github.com/cockroachdb/cockroach/pkg/util/log.(*fileSink).flushAndMaybeSyncLocked()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/file.go:273 +0x16c
  github.com/cockroachdb/cockroach/pkg/util/log.(*fileSink).lockAndFlushAndMaybeSync()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/file.go:235 +0xb5
  github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:87 +0x39
  github.com/cockroachdb/cockroach/pkg/util/log.(*sinkInfoRegistry).iterFileSinks.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/registry.go:84 +0x67
  github.com/cockroachdb/cockroach/pkg/util/log.(*sinkInfoRegistry).iter()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/registry.go:72 +0x145
  github.com/cockroachdb/cockroach/pkg/util/log.(*sinkInfoRegistry).iterFileSinks()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/registry.go:82 +0x52
  github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:86 +0x137

Previous write at 0x00c0003f8268 by goroutine 46:
  bufio.(*Writer).Flush()
      /usr/local/go/src/bufio/bufio.go:619 +0x319
  github.com/cockroachdb/cockroach/pkg/util/log.(*syncBuffer).Flush()
      <autogenerated>:1 +0x49
  github.com/cockroachdb/cockroach/pkg/util/log.(*fileSink).flushAndMaybeSyncLocked()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/file.go:273 +0x16c
  github.com/cockroachdb/cockroach/pkg/util/log.TestRedactedDecodeFile.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/redact_test.go:126 +0x26f
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /usr/local/go/src/testing/testing.go:1306 +0x47

Goroutine 7 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/log.init.5()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/log_flush.go:41 +0x2b

Goroutine 46 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1306 +0x726
  github.com/cockroachdb/cockroach/pkg/util/log.TestRedactedDecodeFile()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/log/redact_test.go:109 +0x32e
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /usr/local/go/src/testing/testing.go:1306 +0x47
==================
    redact_test.go:160: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestRedactedDecodeFile_51108539058
    testing.go:1152: race detected during execution of test
    --- FAIL: TestRedactedDecodeFile/5 (0.01s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
/cc @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRedactedDecodeFile.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/util/log/redact_test.go
+++ b/pkg/util/log/redact_test.go
@@ -123,7 +123,7 @@ func TestRedactedDecodeFile(t *testing.T) {
 			fileName := debugSink.getFileName(t)
 
 			// Ensure our log message above made it to the file.
-			debugSink.flushAndMaybeSyncLocked(false)
+			debugSink.lockAndFlushAndMaybeSync(false)
 
 			// Prepare reading the entries from the file.
 			infoName := filepath.Base(fileName)

